beacon-chain/rpc/beacon/attestations.go

```diff
@@ -237,20 +237,41 @@ func (bs *Server) StreamIndexedAttestations(
 	go bs.collectReceivedAttestations(stream.Context())
 	for {
 		select {
-		case event := <-attestationsChannel:
+		case event, ok := <-attestationsChannel:
+			if !ok {
+				logrus.Error("Stream indexed attestations attestations channel is got closed")
+			}
 			if event.Type == operation.UnaggregatedAttReceived {
 				data, ok := event.Data.(*operation.UnAggregatedAttReceivedData)
 				if !ok {
 					// Got bad data over the stream.
+					logrus.Warningf("Got data of wrong type on stream expected *UnAggregatedAttReceivedData, received %T", event.Data)
 					continue
 				}
 				if data.Attestation == nil {
 					// One nil attestation shouldn't stop the stream.
+					logrus.Info("Got nil attestation")
```

Comment:
this log will show-up at prysm runtime as 

```
[2020-05-11 20:27:06]  INFO blockchain: Processing block state transition...
[2020-05-11 20:27:06]  WARN: Got nil attestation
```
With no context of where it comes from or which function got a nil attestation. I would prefix rpc logs with the RPC prefix like we do in the other packages through a global: `var log = logrus.WithPrefix("rpc")`. Also, would add more informative error messages something like. log.Debug("Indexed attestations stream got a nil attestation") (we should make this one debug level).

Reply:
thanks for the review

Reply:
updated